packages/charts/src/components/TimelineChart/chartbody/TimelineConnections.tsx

```diff
@@ -0,0 +1,329 @@
+import { ThemingParameters } from '@ui5/webcomponents-react-base';
+import React, { ReactElement, useLayoutEffect, useState } from 'react';
+import {
+  ITimelineChartMileStone,
+  ITimelineChartRow,
+  ITimelineChartTask,
+  TimelineChartConnection
+} from '../types/TimelineChartTypes';
+import { ARROWHEAD_HEIGHT, ARROWHEAD_WIDTH, ARROW_CLEARANCE } from '../util/constants';
+
+interface TimelineChartConnectionsProps {
+  dataSet: ITimelineChartRow[];
+
+  /**
+   * The row height. This is used to calculate drawing the
+   * connection arrows.
+   */
+  rowHeight: number;
+
+  /**
+   * The width of the chart body.
+   * This is required to help rerender the arrows when the size
+   * of the chart body changes.
+   */
+  width: number;
+
+  /**
+   * This is required to get the bounding box of the chart body
+   * and to be able to calculate the relative positions of the
+   * start and end points of the connection arrows based on the
+   * bounding box parameters of the chart body.
+   */
+  bodyRect: DOMRect;
+}
+
+interface ConnectionData {
+  startX: number;
+  startY: number;
+  endX: number;
+  endY: number;
+  connection: TimelineChartConnection;
+}
+
+/**
+ * This holds all the arrows that show the connections between different tasks.
+ */
+const TimelineChartConnections = ({ dataSet, width, rowHeight, bodyRect }: TimelineChartConnectionsProps) => {
+  const [connectionDataState, setConnectionDataState] = useState<ConnectionData[]>([]);
+  useLayoutEffect(() => {
+    const connectionData: ConnectionData[] = [];
+
+    const generateConnectionData = (activities: ITimelineChartTask[] | ITimelineChartMileStone[]) => {
```

Comment:
Can this function be moved outside of the component, so it doesn't get recreated every time the effect is executed?